### ISSUE TITLE
Template: conditional params?

### ISSUE THREAD

**Issue by kkmuffme** [vimeo/psalm#9624]
Works (has error): https://psalm.dev/r/aebf79e7cb

Should work too (no error): https://psalm.dev/r/badea121fd

Or is there any other way to make conditional params work correctly?

**Comment by orklah** [comment]
I don't think this was ever supported. Conditionals work only for return and param-out

**Comment by fluffycondor** [comment]
Yeah, totally lacking this feature for typehitting some terrible legacy code.
But at least `@param (T is string ? array : int) $b` works basically as `@param array|int $b`, so you can typehint your code in a conditional way now, it will catch some issues right away, and it will catch some more when the feature will be supported 😊
